src/pybind/mgr/telemetry/tests/telemetry_upgrade_tests.md

```diff
@@ -0,0 +1,208 @@
+waiting for it to complete.
+
+```
+# renamed tasks: to upgrade-sequence:
+upgrade-sequence:
+   sequential:
+   - print: "**** done start upgrade, wait"
+   ...
+       mon.a:
+         - ceph config set global log_to_journald false --force
+         - ceph orch upgrade start --image quay.ceph.io/ceph-ci/ceph:$sha1
+         - while ceph orch upgrade status | jq '.in_progress' | grep true && ! ceph orch upgrade status | jq '.message' | grep Error ; do ceph orch ps ; ceph versions ; ceph orch upgrade status ; sleep 30 ; done
+   ...
+   - print: "**** done end upgrade, wait..."
+```
+
+## Telemetry Upgrade tests
+
+The telemetry upgrade test suite verifies that the telemetry is emitting the
+correct collections after the upgrade. This integration test coverage are done
+via workunits. Workunits are basically bash scripts that run commands against a
+Ceph Cluster. 
+
+In the same manner as the upgrade test, each release branch `qa/workunits`
+directory includes telemetry test for X-2 releases. That means, we can test
+telemetry before and after the upgrade from previous two releases to the current
+one.
+
+For instance, the relevant telemetry workunits for `tentacle` release are:
+```
+qa/workunits
+├── test_telemetry_reef.sh
+├── test_telemetry_reef_x.sh
+├── test_telemetry_squid.sh
+└── test_telemetry_squid_x.sh
+```
+
+- `test_telemetry_reef.sh`, tests the presence of telemetry collection on a reef cluster before the upgrade.
+- `test_telemetry_reef_x.sh`, tests the presence of new telemetry collection on the X-version cluster after it has been upgrade from reef.
+- `test_telemetry_squid.sh`, tests the presence of telemetry collection on a squid cluster before the upgrade
+- `test_telemetry_squid_x.sh`, tests the presence of new telemetry collection on the X-version cluster after it has been upgrade from squid.
+
+A sample telemetry upgrade test file contains the following test: 
+```
+...
+# Assert that new collections are available
+COLLECTIONS=$(ceph telemetry collection ls)
+NEW_COLLECTIONS=("perf_perf" "basic_mds_metadata" "basic_pool_usage"
+                 "basic_rook_v01" "perf_memory_metrics" "basic_pool_options_bluestore")
+for col in ${NEW_COLLECTIONS[@]}; do
+    if ! [[ $COLLECTIONS == *$col* ]];
+    then
+        echo "COLLECTIONS does not contain" "'"$col"'."
+	exit 1
+    fi
+done
+...
+```
+
+These workunits are used in the `upgrade` suite, specifically in:
+- [upgrade/reef-x/parallel](https://github.com/ceph/ceph/tree/main/qa/suites/upgrade/reef-x/parallel)
+- [upgrade/squid-x/parallel](https://github.com/ceph/ceph/tree/main/qa/suites/upgrade/squid-x/parallel)
+- [upgrade/telemetry-upgrade](https://github.com/ceph/ceph/tree/main/qa/suites/upgrade/telemetry-upgrade)
+
+```
+
+upgrade
+├── reef-x
+│   ├── parallel
+│   │   └──  1-tasks.yaml
+├── squid-x
+│   ├── parallel
+│   │   └── 1-tasks.yaml
+└── telemetry-upgrade
+    ├── quincy-x
+    └── reef-x
+
+
+```
```

Comment:
Same as above; update to reflect squid example.

Reply:
thank you! done